File: Go/Day1/1.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("./Go/Day1/1-input.txt")
	if err != nil {
		panic(1)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	left, right := make([]int, 1000), make([]int, 1000)

	for scanner.Scan() {
		line := scanner.Text()
		split := strings.Split(line, "   ")
		left_num, _ := strconv.Atoi(split[0])
		right_num, _ := strconv.Atoi(split[1])
		left = append(left, left_num)
		right = append(right, right_num)
	}

	sort.Sort(sort.IntSlice(left))
	sort.Sort(sort.IntSlice(right))

	total := 0
	for i := range len(left) {
		dif := math.Abs(float64(left[i] - right[i]))
		total += int(dif)
	}

	fmt.Println(total)
}

File: Python/Day1/1.py
import os

left, right, raw = [], [], []

with open("./Python/Day1/1-input.txt", "r") as f:
    raw = f.readlines()

for line in raw:
    split_str = line.split("   ")
    left.append(int(split_str[0]))
    right.append(int(split_str[1]))

left.sort()
right.sort()

total = 0
for i in range(len(left)):
    total += abs(left[i] - right[i])

print(total)